packages/zone/src/durable.js

```diff
@@ -55,31 +58,49 @@ export const detachedDurableStores = attachDurableStores(() =>
  * Create a zone whose objects persist between Agoric vat upgrades.
  *
  * @param {import('@agoric/vat-data').Baggage} baggage
+ * @param {string} [baseLabel]
  * @returns {import('.').Zone}
  */
-export const makeDurableZone = baggage => {
+export const makeDurableZone = (baggage, baseLabel = 'durableZone') => {
   baggage || Fail`baggage required`;
+
+  const attachedStores = attachDurableStores(() => baggage);
+
+  const { makeOnce, wrapProvider } = makeOnceKit(
+    baseLabel,
+    attachedStores,
+    baggage,
+  );
+
   /** @type {import('.').Zone['exoClass']} */
   const exoClass = (...args) => prepareExoClass(baggage, ...args);
   /** @type {import('.').Zone['exoClassKit']} */
   const exoClassKit = (...args) => prepareExoClassKit(baggage, ...args);
   /** @type {import('.').Zone['exo']} */
   const exo = (...args) => prepareExo(baggage, ...args);
 
-  const attachedStores = attachDurableStores(() => baggage);
+  const mapStore = wrapProvider(attachedStores.mapStore);
 
   /** @type {import('.').Zone['subZone']} */
   const subZone = (label, options = {}) => {
-    const subBaggage = provideDurableMapStore(baggage, label, options);
-    return makeDurableZone(subBaggage);
+    const subBaggage = mapStore(label, options);
+    return makeDurableZone(subBaggage, `${baseLabel}.${label}`);
   };
 
   return Far('durableZone', {
-    exo,
-    exoClass,
-    exoClassKit,
+    exo: wrapProvider(exo),
+    exoClass: wrapProvider(exoClass),
+    exoClassKit: wrapProvider(exoClassKit),
     subZone,
-    ...attachedStores,
+
+    makeOnce,
+    detached: attachedStores.detached,
```

Comment:
(out of topic for this PR but notice while reviewing)

Above, we have:

```js
export const detachedDurableStores = attachDurableStores(() =>
  makeScalarMapStore('detached'),
);
```

which means that each maker on the `detached` stores will use a separately allocated "detached" fake baggage. That doesn't seem correct? Maybe we need some memoization somewhere?

Reply:
This was the only way I could figure out how to use the existing `provideDurable*Store` functions.  Otherwise, using the same label for different detached stores would throw when the key is put into the fake baggage.

Fixing this properly would require refactoring the underlying layers to expose their innards.  I hope that can be done, but as you said, not now.

Reply:
I agree that no change for this is needed in this PR. Resolving.